src/components/Output/subcomponents/NitrogenWidget.jsx

```diff
@@ -137,8 +139,18 @@ const NitrogenCard = ({ refVal }) => {
   useEffect(() => {
     if (chartRef1.current && chartRef2.current) {
       if (!surfaceData || surfaceData.length === 0) {
-        chartRef1.current.chart.showLoading();
-        chartRef2.current.chart.showLoading();
+        const spinnerHtml = ReactDOMServer.renderToString(
```

Comment:
why is this needed on the spinner component?

Reply:
@ravighevariya10

Reply:
It's needed because showLoading() expects an HTML string to be passed and to integrate our shared loading-indicator, we have to pass it as string HTML so this 'ReactDOMServer.renderToString()' function converts the React component to an HTML string